src/streaming/rules/ThroughputHistory.js

```diff
@@ -103,6 +119,19 @@ function ThroughputHistory(config) {
         if (latencyDict[mediaType].length > MAX_MEASUREMENTS_TO_KEEP) {
             latencyDict[mediaType].shift();
         }
+
+        updateEwmaEstimate(ewmaThroughputDict[mediaType], throughput, 0.001 * downloadTimeInMilliseconds, ewmaHalfLife.throughputHalfLife);
+        updateEwmaEstimate(ewmaLatencyDict[mediaType], latencyTimeInMilliseconds, 1, ewmaHalfLife.latencyHalfLife);
+    }
+
+    function updateEwmaEstimate(ewmaObj, value, weight, halfLife) {
+        const fastAlpha = Math.pow(0.5, weight / halfLife.fast);
+        ewmaObj.fastEstimate = (1 - fastAlpha) * value + fastAlpha * ewmaObj.fastEstimate;
```

Comment:
So at startup, ewmaObj.fastEstimate is 0. So this is underestimating the EWMA at the beginning,(and as time rolls on, the 0 is weighted to almost nothing). Instead, the very first EWMA value should be just the first measurement. So, maybe we should be doing something like: set ewmaObj.fastEstimate to undefined, when the first measurement comes in, set ewmaObj.fastEstimate to that value, then carry on calculating the EWMA as before for future measurements.